data-raw/Station Data prep.R

```diff
@@ -0,0 +1,162 @@
+
+
+
+## Integrated water quality ------------------------------------------------
+
+WQ<-wq(Start_year = 2015, End_year = 2021,
+       Sources = c("EMP", "USGS_SFBS", "USGS_CAWSC"))%>%
+  filter(!is.na(Latitude) & !is.na(Longitude), !str_detect(Station, "EZ"))%>%
+
+  group_by(Year, Station,  Source)%>%
+  mutate(n_Lat=n_distinct(Latitude),
+         n_Lon=n_distinct(Longitude))%>%
+  ungroup()%>%
+  group_by(Station, Source)%>%
+  summarise(Latitude=unique(Latitude), Longitude=unique(Longitude), N=n(), .groups="drop")%>%
+  mutate(Parameter="Water_quality")
+
+
+## Fish --------------------------------------------------------------------
+
+
+Fish<-wq(Start_year = 2015, End_year = 2021,
+         Sources = c("STN", "FMWT", "DJFMP", "SKT", "20mm", "Suisun",
+                     "Baystudy", "SLS"))%>%
+  bind_rows(YBFMP_fish)%>%
+  mutate(Parameter= case_when(Source == "FMWT" ~ "MDWT",
+                              Source == "Suisun" ~ "Otter Trawl",
+                              Source == "Baystudy" ~ "MDWT and Otter",
+                              Source == "SKT" ~ "KDT",
+                              Source == "SLS" ~ "Larval Sled",
+                              Source == "20 mm" ~ "Sled",
+                              Source == "STN" ~ "Sled"))
+
+# Benthic and phytoplankton ----------------------------------------------
+
+EMPbenthic = read_csv("https://portal.edirepository.org/nis/dataviewer?packageid=edi.1036.3&entityid=4e6948186ad756dc2b6de4de41b601f3")
+
+benthic_stations = filter(EMPbenthic, Period_of_Record_To == "Present") %>%
+  rename(Station = StationCode, StationName = Location) %>%
+  select(Station, StationName, Latitude, Longitude) %>%
+  mutate(Parameter = "Benthic", Source = "EMP")
+
+## Phytoplankton ----------------------------------------------------------------
+
+phyto = read_csv("https://portal.edirepository.org/nis/dataviewer?packageid=edi.1320.4&entityid=00b61678378071f779f1135be223fe0a")
+
+phyto_stations = rename(phyto, Station = StationCode, StationName = Location) %>%
+  select(Station, StationName, Latitude, Longitude) %>%
+  mutate(Parameter = "Phytoplankton", Source = "EMP")
+
+cdec = read_csv("data-raw/CDEC_stations.csv") %>%
+  rename(Station = STA, StationName = `Station Name`) %>%
+  mutate(Parameter = "Cont_Water_Quality", Source = "CDEC") %>%
+  select(Station, StationName, Latitude, Longitude, Parameter, Source)
+
+# Merge everything --------------------------------------------------------
+
+Stations<-bind_rows(benthic_stations, phyto_stations, Zoop, WQ, Fish, cdec) %>%
+  filter(!is.na(Latitude)) %>%
+  select(Station, StationName, Latitude, Longitude, Parameter, Source) %>%
+  distinct() %>%
+  st_as_sf(coords = c("Longitude", "Latitude"), crs = 4326, remove = F) %>%
+  st_transform(crs = st_crs(deltamapr::R_Delta))
+
+# #test with a quick map
+# ggplot()+
+#   geom_sf(data = deltamapr::WW_Delta)+
+#   geom_sf(data = Stations, aes(color = Source))
+
+save(Stations, file="data/Stations.rda")
```

Comment:
Can you use the usethis::use_data function for this as it is used in the other data-raw scripts?

Reply:
Done!